src/irr/asset/CPLYMeshFileLoader.cpp

```diff
@@ -261,41 +261,42 @@ asset::SAssetBundle CPLYMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 				}
 			}
 
+			asset::SBufferBinding<ICPUBuffer> bufferBinding;
+
             if (indices.size())
             {
-				ICPUMeshBuffer::SBufferBinding bufferBinding;
 				bufferBinding.buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(indices.size() * sizeof(uint32_t));
+				bufferBinding.offset = 0ull;
                 memcpy(bufferBinding.buffer->getPointer(), indices.data(), bufferBinding.buffer->getSize());
 
-				if (!genVertBuffersForMBuffer(mb.get(), attribs, bufferBinding))
-					return {};
+				mb->setIndexBufferBinding(std::move(bufferBinding));
 
                 mb->setIndexCount(indices.size());
                 mb->setIndexType(asset::EIT_32BIT);
 				mb->getPipeline()->getPrimitiveAssemblyParams().primitiveType = E_PRIMITIVE_TOPOLOGY::EPT_TRIANGLE_LIST;
+
+				if (!genVertBuffersForMBuffer(mb.get(), attribs, bufferBinding))
```

Comment:
you've fucked up, did you even test?

`bufferBinding` will now be `nullptr` because you've moved it above?

Reply:
yea that true, but `genVertBuffersForMBuffer` should not actually deal with indices, so to fix a little bit